mri/reconstruct/fourier.py

```diff
@@ -70,66 +70,105 @@ def adj_op(self, x):
 
 
-class FFT2(FourierBase):
-    """ Standard unitary 2D Fast Fourrier Transform class.
-    The FFT2 will be normalized in a symmetric way
+class FFT(FourierBase):
+    """ Standard unitary ND Fast Fourrier Transform class.
+    The FFT will be normalized in a symmetric way
 
     Attributes
     ----------
     samples: np.ndarray
         the mask samples in the Fourier domain.
     shape: tuple of int
         shape of the image (not necessarly a square matrix).
+     n_coils: int, default 1
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition. If n_coils > 1, data shape must be
+            [n_coils, Nx, Ny, NZ]
     """
-    def __init__(self, samples, shape):
-        """ Initilize the 'FFT2' class.
+    def __init__(self, samples, shape, n_coils=1):
+        """ Initilize the 'FFT' class.
 
         Parameters
         ----------
         samples: np.ndarray
             the mask samples in the Fourier domain.
         shape: tuple of int
             shape of the image (not necessarly a square matrix).
+         n_coils: int, default 1
+                Number of coils used to acquire the signal in case of
+                multiarray receiver coils acquisition. If n_coils > 1,
+                 data shape must be equal to [n_coils, Nx, Ny, NZ]
         """
         self.samples = samples
         self.shape = shape
         self._mask = convert_locations_to_mask(self.samples, self.shape)
+        if n_coils <= 0:
+            warn("The number of coils should be strictly positive")
+            n_coils = 1
+        self.n_coils = n_coils
 
     def op(self, img):
-        """ This method calculates the masked Fourier transform of a 2-D image.
+        """ This method calculates the masked Fourier transform of a ND image.
 
         Parameters
         ----------
         img: np.ndarray
-            input 2D array with the same shape as the mask.
+            input ND array with the same shape as the mask. For multichannel
+            images the coils dimension is put first
 
         Returns
         -------
         x: np.ndarray
-            masked Fourier transform of the input image.
+            masked Fourier transform of the input image. For multichannel
+            images the coils dimension is put first
         """
-        return self._mask * np.fft.fft2(img, norm="ortho")
+        if self.n_coils == 1:
+            return self._mask * np.fft.ifftshift(np.fft.fftn(
+                                    np.fft.fftshift(img), norm="ortho"))
+        else:
+            if self.n_coils > 1 and self.n_coils != img.shape[0]:
+                raise ValueError("The number of coils parameter is not equal"
+                                 "to the actual number of coils, the data must"
+                                 "be reshaped as [n_coils, Nx, Ny, Nz]")
+            else:
+                return np.asarray([self._mask * np.fft.ifftshift(np.fft.fftn(
```

Comment:
Do we have speedup if we use joblib and run on multiple CPUs? Just checking. Unlike NFFT, if this gives significant speedup, it would be good to expose it to user.

Reply:
Make it as an enhancement